src/subscription/subscribe.js

```diff
@@ -220,16 +249,23 @@ export function createSourceEventStream(
     info
   );
 
-  if (subscription instanceof Error) {
-    throw subscription;
-  }
+  return Promise.resolve(subscription).then(resolvedSubscription => {
```

Comment:
I added a test case that validates bad query variables are returned as an ExecutionResult. However, the `return new Promise` approach you suggested led to pretty gnarly logic for dealing with the nested promise coming back from resolveFieldValueOrError. The current version uses await/async to work around it, but happy to revise if you see don't think the tradeoff in code clarity is worth it.